diff --git a/Cargo.toml b/Cargo.toml
index 8a6afa4..a793fbc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -208,8 +208,8 @@ k256 = { git = "https://github.com/powdr-labs/openvm.git", package = "k256", tag
 p256 = { git = "https://github.com/powdr-labs/openvm.git", package = "p256", tag = "v2.0.0-beta.2-powdr.1" }
 
 [patch."https://github.com/openvm-org/stark-backend.git"]
-openvm-cpu-backend = { git = "https://github.com/powdr-labs/stark-backend.git", tag = "v2.0.0-beta.2-powdr" }
-openvm-cuda-backend = { git = "https://github.com/powdr-labs/stark-backend.git", tag = "v2.0.0-beta.2-powdr" }
-openvm-cuda-common = { git = "https://github.com/powdr-labs/stark-backend.git", tag = "v2.0.0-beta.2-powdr" }
-openvm-stark-backend = { git = "https://github.com/powdr-labs/stark-backend.git", tag = "v2.0.0-beta.2-powdr" }
-openvm-stark-sdk = { git = "https://github.com/powdr-labs/stark-backend.git", tag = "v2.0.0-beta.2-powdr" }
+openvm-cpu-backend = { path = "../../stark-backend-autoopt/crates/cpu-backend" }
+openvm-cuda-backend = { path = "../../stark-backend-autoopt/crates/cuda-backend" }
+openvm-cuda-common = { path = "../../stark-backend-autoopt/crates/cuda-common" }
+openvm-stark-backend = { path = "../../stark-backend-autoopt/crates/stark-backend" }
+openvm-stark-sdk = { path = "../../stark-backend-autoopt/crates/stark-sdk" }
